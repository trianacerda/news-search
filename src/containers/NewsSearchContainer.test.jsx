import React from 'react';
import { render, screen } from '@testing-library/react';
import NewsSearchContainer from './NewSearchContainer';
// import { rest } from 'msw';
// import { setupServer } from 'msw/node';
// const server = setupServer(
//   // Describe the requests to mock.
//   rest.get(
//     `https://newsapi.org/v2/everything?q=bitcoin&apiKey=${process.env.REACT_APP_API_KEY}`,
//     (req, res, ctx) => {
//       rest.get('https://newsapi.org/v2/everything', (req, res, ctx) => {
//         const query = req.url.searchParams;
//         const q = query.get('q');
//         const apiKey = query.get('apiKey');
//       });
//       return res(
//         ctx.json({
//           title:
//             'JPMorgan CEO Says Bitcoin Has No Intrinsic Value After Claiming That Its Price Could Rise 10X',
//           author: 'noreply@blogger.com (Unknown)',
//           description:
//             'JPMorgan CEO Jamie Dimon is still not a Bitcoin fan despite rapid price appreciation   from U.Today https://ift.tt/3iyi5nR https://ift.tt/eA...',
//         })
//       );
//     }
//   )
// );

describe('News Search Testing', () => {
  // beforeAll(() => {
  //   // Establish requests interception layer before all tests.
  //   server.listen();
  // });
  // afterAll(() => {
  //   // Clean up after all tests are done, preventing this
  //   // interception layer from affecting irrelevant tests.
  //   server.close();
  // });

  it('should display a list of articles', async () => {
    render(<NewsSearchContainer />);
    screen.getByText('Loading...');

    // const ul = await screen.findByRole('list', { name: 'articles' });
    // expect(ul).toMatchSnapshot();
  });
});
