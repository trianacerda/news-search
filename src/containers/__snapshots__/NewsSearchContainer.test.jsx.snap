// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News Search Testing should display a list of articles 1`] = `
<ul
  aria-label="articles"
>
  <li>
    <figure>
      <figcaption>
        Facebook sues programmer who allegedly scraped data for 178 million users
        Jon Fingas
        Facebook is taking legal action in response to another large-scale data heist. According to The Record, the social network has sued Ukraine national Alexander Solonchenko for allegedly scraping data for more than 178 million users. Solonchenko reportedly expl…
      </figcaption>
    </figure>
  </li>
  <li>
    <figure>
      <figcaption>
        Yoko Taro Foresees the End of Video Games As We Know Them
        Joshua Khan
        The mind behind iconic titles like 'Nier: Automata' and 'Drakengard' tells WIRED about his inspirations—and how AI plays a role in the future of game design.
      </figcaption>
    </figure>
  </li>
  <li>
    <figure>
      <figcaption>
        How Googlers are making unseen disability visible at work
        Sneha SrinivasanProgram Manager, Google India
        Five years ago, I discovered that I had left temporal epilepsy, a condition that causes unprovoked seizures. To manage it, I was prescribed medication. While the medication controls the brain activity, the downside was experiencing fatigue, long sleep cycles,…
      </figcaption>
    </figure>
  </li>
  <li>
    <figure>
      <figcaption>
        限时优惠：亚马逊 Chromebook 特价，最低七折
        Eric Chan
        Chromebook 的低價、易用、快速，是不少低預算但卻又有一定要求的用戶，尤其是學生的歡迎。Lenovo 的熱門機型 Chromebook Flex 5 13 吋，在今天就來一個限時優惠！
      </figcaption>
    </figure>
  </li>
  <li>
    <figure>
      <figcaption>
        Matthew McConaughey, who's called himself a 'folk singin', philosopher, poet-statesman,' says he's not sure he wants to run for Texas governor because politics is 'a bag of rats'
        aharoun@businessinsider.com (Azmi Haroun)
        "It's necessary to be aggressively centric to salvage democracy in America right now," McConaughey said, stressing the need for a third party.
      </figcaption>
    </figure>
  </li>
</ul>
`;
